pandas/core/groupby.py

```diff
@@ -1923,8 +1923,16 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
 
             # a passed Categorical
             elif isinstance(self.grouper, Categorical):
+
+                # must have an ordered categorical
+                if self.sort:
+                    if not self.grouper.ordered:
+                        raise ValueError("cannot sort by an unordered Categorical in the grouper\n"
```

Comment:
- on the first ``df['A'].order()``, that is correct. The ``.order()`` raises telling the user they were relying on an implicity ordered Categorical
- ``sort=True`` is the *default* for grouping, so by definition is *should* raise. The user was using an implicitly ordered Categorical in prior versions, that is why I am noting it